sparkle/src/components/ElementDialog.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { useCallback, useState } from "react";
+
+import { Dialog, ModalProps } from "./Dialog";
+
+type ElementDialogProps<T> = Omit<
+  ModalProps,
+  "isOpen" | "onValidate" | "onCancel"
+> & {
+  openOnElement: T | null;
+  onValidate: (closeDialogFn: () => void) => void;
```

Comment:
This patterns feel weird, like the `Dialog` is not a pure component anymore 🤔.

Reply:
Yes the component now has a state (transitioning close) so it's indeed not pure anymore

I agree it feels a bit weird, but IMHO better than having to have multiple states + setTimeouts each time we have Element dialogs. 

I'll refactor a few modals, you'll see it looks better

Reply:
💯